controllers/usersignup/usersignup_controller.go

```diff
@@ -112,6 +118,51 @@ func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.
 		return reconcile.Result{}, nil
 	}
 
+	// REMOVE THIS BLOCK AFTER MIGRATION - See additional comments further below in this function
+	originalUserSignupName, ok := userSignup.Annotations[migrationAnnotationName]
+	if ok {
+		// If the annotation has been set, then it indicates that this UserSignup has been migrated, and that the
+		// *original* UserSignup should be deleted.  The value of the annotation is the name of the original UserSignup
+		// resource that should now be deleted.
+		//
+		// This block of migration code should:
+		//
+		// 1) Delete the original UserSignup, then
+		// 2) Remove the migration-replaces annotation
+		userSignupToDelete := &toolchainv1alpha1.UserSignup{}
+		err = r.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: request.Namespace,
+			Name:      originalUserSignupName,
+		}, userSignupToDelete)
+		if err != nil {
+			// If the annotation exists however the original UserSignup isn't found, then remove the annotation
+			if errors.IsNotFound(err) {
+				// Remove the annotation
+				delete(userSignup.Annotations, migrationAnnotationName)
+				// Update the UserSignup
+				err = r.Client.Update(ctx, userSignup)
+				if err != nil {
+					return reconcile.Result{}, r.wrapErrorWithStatusUpdate(logger, userSignup,
+						r.setStatusMigrationFailedCleanup, err, fmt.Sprintf("Failed to remove migration annotation"))
+				}
```

Comment:
We should either return here after successfully deleting the annotation (but you would need to update the predicate to watch the migration annotation too: https://github.com/codeready-toolchain/host-operator/blob/95d7140dd246549775d077cfd1e1b039f6678187/controllers/usersignup/predicate.go#L24-L41) or skip the rest of the migration code (do not try to delete the original usersignup below) and proceed with the reconcile.

Reply:
Restructured this so that the delete occurs up front